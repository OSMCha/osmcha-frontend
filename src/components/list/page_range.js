// @flow
import React from 'react';

export class PageRange extends React.PureComponent {
  render() {
    return (
      <button
        onClick={this._onClick}
        disabled={this.props.disabled}
        className={`flex-child btn btn--s color-gray round bg-gray-faint bg-darken10-on-active bg-darken5-on-hover
          ${this.props.active && 'is-active'}
          `}
      >
        {typeof this.props.page === 'number'
          ? this.props.page + 1
          : <svg className="icon icon--s inline-block align-middle ">
              <use xlinkHref={`#icon-${this.props.page}`} />
            </svg>}
      </button>
    );
  }
  _onClick = () => {
    this.props.getChangesetsPage(this.props.pageIndex);
  };
}
