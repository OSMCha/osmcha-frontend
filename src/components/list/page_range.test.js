import React from 'react';
import { PageRange } from './page_range';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <PageRange
        page={'arrow-left'}
        pageIndex={0}
        disabled={false}
        active={false}
        getChangesetsPage={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <PageRange
        key={1}
        page={4}
        pageIndex={4}
        active={true}
        getChangesetsPage={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
