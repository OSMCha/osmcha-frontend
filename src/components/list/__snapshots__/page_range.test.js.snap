// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="flex-child btn btn--s color-gray round bg-gray-faint bg-darken10-on-active bg-darken5-on-hover
          false
          "
  disabled={false}
  onClick={[Function]}
>
  <svg
    className="icon icon--s inline-block align-middle "
  >
    <use
      xlinkHref="#icon-<"
    />
  </svg>
</button>
`;

exports[`renders correctly 2`] = `
<button
  className="flex-child btn btn--s color-gray round bg-gray-faint bg-darken10-on-active bg-darken5-on-hover
          is-active 
          "
  onClick={[Function]}
>
  5
</button>
`;
